nixos/modules/system/boot/zram-as-tmp.nix

```
@@ -0,0 +1,96 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.boot.tmp;
+in
+{
+  options = {
+    boot.tmp = {
+      useZram = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        example = true;
+        description = ''
+           Whether to mount a zram device on {file}`/tmp` during boot.
+
+           ::: {.note}
+           Large Nix builds can fail if the mounted zram device is not large enough.
+           In such a case either increase the zramSettings.zram-size or disable this option.
+           :::
+        '';
+      };
+
+      zramSettings = {
+        zram-size = mkOption {
+          type = types.str;
```

```suggestion
          type = lib.types.str;
```
Same for other types